### ISSUE TITLE
SunOS (10) - EINPROGRESS in pyuv.Async during TCP.connect causes exception 

### ISSUE THREAD

**Issue by alxchk** [saghul/pyuv#237]
```import pyuv
import threading
import Queue
import logging

class UVThread(threading.Thread):
    def __init__(self):
        self.loop = pyuv.Loop()
        self.queue = Queue.Queue()
        self.wake = pyuv.Async(self.loop, self.sync)
        super(UVThread, self).__init__()

    def sync(self, handle):
        while True:
            try:
                method, args = self.queue.get_nowait()
            except Queue.Empty:
                break

            try:
                method(*args)
            except Exception, e:
                logging.exception('Defered call exception: {}'.format(e))

    def defer(self, method, *args):
        self.queue.put((method, args))
        self.wake.send()

    def connect(self, addr):
        c = pyuv.TCP(self.loop)
        print "Connect: ", addr
        try:
            c.connect(addr, self.on_connect)
        except Exception, e:
            print "Exception: ", e
            self.loop.stop()
            c.close()

    def on_connect(self, handle, error):
        print "CONNECTED: ", handle, error

    def run(self):
        self.loop.run()

uv = UVThread()
uv.start()

uv.defer(uv.connect, ('192.168.0.1', 80))
```
```
bash-4.3# python test.py
Connect:  ('192.168.0.1', 80)
Exception:  (-11, 'resource temporarily unavailable')
```

```import pyuv

def on_connect(handle, error):
    print handle, error

loop = pyuv.Loop()
c = pyuv.TCP(loop)
c.connect(('192.168.0.1', 80), on_connect)
loop.run()
```

```Connect:  /2:   write(1, " C o n n e c t :    ", 10)            = 10
('192.168.0.1', 80)
/2:     write(1, " ( ' 1 9 2 . 1 6 8 . 0 .".., 20)      = 20
/2:     so_socket(PF_INET, SOCK_STREAM, IPPROTO_IP, "", SOV_XPG4_2) = 11
/2:     fcntl(11, F_GETFL)                              = 2
/2:     fcntl(11, F_SETFL, FWRITE|FNONBLOCK)            = 0
/2:     fcntl(11, F_GETFD, 0x00000082)                  = 0
/2:     fcntl(11, F_SETFD, 0x00000001)                  = 0
/2:     connect(11, 0xFE529650, 16, SOV_XPG4_2)         Err#150 EINPROGRESS
Exception:  /2: write(1, " E x c e p t i o n :    ", 12)        = 12
(-11, 'resource temporarily unavailable')
/2:     write(1, " ( - 1 1 ,   ' r e s o u".., 42)      = 42
```

Without async all works as expected:
```bash-4.3# python test2.py
<pyuv._cpyuv.TCP object at 0xfead07ac> None
```

```so_socket(PF_INET, SOCK_STREAM, IPPROTO_IP, "", SOV_XPG4_2) = 11
fcntl(11, F_GETFL)                              = 2
fcntl(11, F_SETFL, FWRITE|FNONBLOCK)            = 0
fcntl(11, F_GETFD, 0x00000082)                  = 0
fcntl(11, F_SETFD, 0x00000001)                  = 0
connect(11, 0x08046C10, 16, SOV_XPG4_2)         Err#150 EINPROGRESS
port_associate(5, 4, 0x00000006, 0x00000001, 0x00000000) = 0
port_associate(5, 4, 0x00000008, 0x00000001, 0x00000000) = 0
port_associate(5, 4, 0x0000000B, 0x00000004, 0x00000000) = 0
port_getn(5, 0x08042CB0, 1024, 1, 0x00000000)   = 1 [0]
getsockopt(11, SOL_SOCKET, SO_ERROR, 0x08042BF0, 0x08042BE8, SOV_XPG4_2) = 0
ioctl(1, TCGETA, 0x08042654)                    = 0
fstat64(1, 0x080425C0)                          = 0
<pyuv._cpyuv.TCP object at 0xfead07ac> write(1, " < p y u v . _ c p y u v".., 39)       = 39
```

**Comment by alxchk** [comment]
This error doesn't happens with deps/libuv bumped to latest git

**Comment by schlamar** [comment]
@alxchk Can you please check the current v1.x branch, this issue should be fixed in #241
